File: Java/Ram.java
public class Ram extends Komponen {
    private int kapasitasGB;
    private String tipe;     // DDR4, DDR5, etc
    private int frekuensiMHz;

    public Ram() {
        super();
        this.kapasitasGB = 0;
        this.tipe = "";
        this.frekuensiMHz = 0;
    }

    public Ram(int kapasitasGB, String tipe, int frekuensiMHz, String merk, String nama, int harga) {
        super(merk, nama, harga);
        this.kapasitasGB = kapasitasGB;
        this.tipe = tipe;
        this.frekuensiMHz = frekuensiMHz;
    }

    public void setKapasitasGB(int kapasitasGB) {
        this.kapasitasGB = kapasitasGB;
    }

    public void setTipe(String tipe) {
        this.tipe = tipe;
    }

    public void setFrekuensiMHz(int frekuensiMHz) {
        this.frekuensiMHz = frekuensiMHz;
    }

    public int getKapasitasGB() {
        return this.kapasitasGB;
    }

    public String getTipe() {
        return this.tipe;
    }

    public int getFrekuensiMHz() {
        return this.frekuensiMHz;
    }

    @Override
    public void displayInfo() {
        super.displayInfo();
        System.out.println("Kapasitas: " + this.kapasitasGB + " GB");
        System.out.println("Tipe: " + this.tipe);
        System.out.println("Frekuensi: " + this.frekuensiMHz + " MHz");
    }
}

File: Cpp/Ram.cpp
#pragma once
#include <iostream>
#include <string>
#include "Komponen.cpp"
using namespace std;

class Ram : public Komponen {
private:
    int kapasitasGB;
    string tipe;     // DDR4, DDR5, etc
    int frekuensiMHz; // tambahan atribut

public:
    Ram() : Komponen() {
        this->kapasitasGB = 0;
        this->tipe = "";
        this->frekuensiMHz = 0;
    }

    Ram(int kapasitasGB, string tipe, int frekuensiMHz, string merk, string nama, int harga) 
        : Komponen(merk, nama, harga) {
        this->kapasitasGB = kapasitasGB;
        this->tipe = tipe;
        this->frekuensiMHz = frekuensiMHz;
    }

    void setKapasitasGB(int kapasitasGB) {
        this->kapasitasGB = kapasitasGB;
    }

    void setTipe(string tipe) {
        this->tipe = tipe;
    }

    void setFrekuensiMHz(int frekuensiMHz) {
        this->frekuensiMHz = frekuensiMHz;
    }

    int getKapasitasGB() const {
        return this->kapasitasGB;
    }

    string getTipe() const {
        return this->tipe;
    }

    int getFrekuensiMHz() const {
        return this->frekuensiMHz;
    }

    void displayInfo() const override {
        Komponen::displayInfo();
        cout << "Kapasitas: " << this->kapasitasGB << " GB" << endl;
        cout << "Tipe: " << this->tipe << endl;
        cout << "Frekuensi: " << this->frekuensiMHz << " MHz" << endl;
    }

    ~Ram() {}
};
